### PR TITLE
syncer: table no need exist when drop

### LINKED ISSUE #975: Fail to drop not exist table.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
`DROP TABLE IF NOT EXISTS table_name;` while table not exists.
2. What did you expect to see?
DM work normally.
3. What did you see instead?
```
"errors": [
{
    "ErrCode": 44007,
    "ErrClass": "schema-tracker",
    "ErrScope": "downstream",
    "ErrLevel": "medium",
    "Message": "startLocation: [position: (test-e108ea4b-e2f8-4ed1-8ca3-e84a4af392c4-j3ds7-k4j1b-bin.000008, 11865), gtid-set: e01311c8-ec19-11ea-9bc6-be6e9048b4d1:1-443], endLocation: [position: (test-e108ea4b-e2f8-4ed1-8ca3-e84a4af392c4-j3ds7-k4j1b-bin.000008, 11999), gtid-set: e01311c8-ec19-11ea-9bc6-be6e9048b4d1:1-443]: cannot fetch downstream table schema of `ha_test`.`ta` to initialize upstream schema `ha_test`.`ta` in schema tracker",
    "RawCause": "Error 1146: Table 'ha_test.ta' doesn't exist",
    "Workaround": ""
}
```
4. Versions of the cluster

    - DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):

        ```
        master
        ```
This may also happen when replicate `drop table statement` twice even in safemode. see https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6476/pipeline/

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
close #975 

### What is changed and how it works?
- do not check table exist when drop table because we have added `IF EXIST` for  DROP TABLE statement.

Tests 

 - Integration test

### CODE REVIEW

File: tests/all_mode/run.sh [hunk 1]
```diff
@@ -159,6 +159,16 @@ function run() {
     check_log_not_contains $WORK_DIR/worker1/log/dm-worker.log "123456"
     check_log_not_contains $WORK_DIR/worker2/log/dm-worker.log "123456"
 
+    # test drop table if exists
+    run_sql_source1 "drop table if exists \`all_mode\`.\`tb1\`;"
+    run_sql_source1 "drop table if exists \`all_mode\`.\`tb1\`;"
+    run_sql_source2 "drop table if exists \`all_mode\`.\`tb2\`;"
+    run_sql_source2 "drop table if exists \`all_mode\`.\`tb2\`;"
+    sleep 2
```
Comment (lance6716) [inline tests/all_mode/run.sh:167]: how about retry and check downstream table dropped, maybe quicker than 2s

**Review by lance6716 — COMMENTED** [review]
rest LGTM

File: tests/all_mode/run.sh (hunk 1 above)
Comment (GMHDBJD) [reply tests/all_mode/run.sh:167]: I want to check no error exist.

Comment (lance6716) [reply tests/all_mode/run.sh:167]: just want to replace sleep 2s this line 🤣

Comment (GMHDBJD) [reply tests/all_mode/run.sh]: check log in 5bb5f63

File: syncer/syncer.go [hunk 2]
```diff
@@ -2100,8 +2100,6 @@ func (s *Syncer) trackDDL(usedSchema string, sql string, tableNames [][]*filter.
 		shouldSchemaExist = true
 	case *ast.DropTableStmt:
 		shouldExecDDLOnSchemaTracker = true
-		shouldSchemaExist = true
```
Comment (csuzhangxc) [inline syncer/syncer.go:2103]: Do we need to handle `shouldSchemaExist` for `case *ast.DropDatabaseStmt`?

**Review by csuzhangxc — COMMENTED** [review]
rest LGTM

File: syncer/syncer.go (hunk 2 above)
Comment (GMHDBJD) [reply syncer/syncer.go:2103]: We create it if not exist below. But actually no need. Address in 56bf87f

**Comment by lance6716** [comment]
LGTM

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**